discovery.md

```diff
@@ -101,18 +101,295 @@ be considered an event subscriber and a target user of this API.
+
+### REST Paths
+
+Each path in the REST API represents either represents a list (or search) over
+and entity class, or the retrieval of an individual entity by name. Each of
+these operations MUST be supported by compliant discovery providers.
+
+#### `/types?matching=[search term]`
+
+Returns a list of all types in the discovery system, optionally filtering on
+a provided search term (`matching`).
+
+##### matching
+
+* Type: `string`
+* Description: Search term that provides case insensitive match against `type`
+  names. The parameter can match any portion of the type name.
+* Constraints:
+  * OPTIONAL
+  * If present, must be non-empty
+* Examples:
+  * `"com.storage.object"`
+    * matches:
+      * `"com.storage.object.create"`
+      * `"com.storage.object.delete"`
+      * `"com.storage.object.update"`
+
+##### Returns
+
+A JSON array of `type` entities.
+
+#### `/types/{name}`
+
+Retrieves an individual type entity by exact match on the `type` name.
+
+##### Returns
+
+A `type` entity as a JSON object.
+
+#### `/types/{name}/providers`
+
+Retrieves the details about providers that offer the specified `type`.
+
+##### Returns
+
+A JSON array of `provider` entities.
+
+#### `/providers?matching=[search term]`
+
+Returns a list of all providers int he discovery system, optionally filtering on
+a provided search term (`matching`).
+
+##### matching
+
+* Type: `string`
+* Description: Search term that provides case insensitive match against
+  `provider` names. The parameter can match any portion of the provider name.
+* Constraints:
+  * OPTIONAL
+  * If present, must be non-empty
+* Examples:
+  * `"storage"`
+    * matches:
+      * `"Awesome Cloud Storage"`
+      * `"File storage system"`
+      * `"storage Storage STORAGE"`
+
+##### Returns
+
+A JSON array of `provider` entities.
+
+#### `/providers/{name}`
+
+Retrieves an individual provider entity by exact match on the `provider` name.
+
+##### Returns
+
+A `provider` entity as a JSON object.
+
+#### `/providers/{name}/types`
+
+Retrieves the details about the types that are offered by the specified
+`provider`.
+
+##### Returns
+
+A JSON array of `type` entities
+
+#### `/producer/{provider.name}/{type.name}`
+
+Retrieves the `producer` entity that specifies the information necessary to
+create subscriptions and to consume the events. The `provider.name` and
+`type.name` items in the request path correspond to the information that can
+be obtained via the `/provider` and `/type` API calls.
+
+##### Returns
+
+A `producer` entity as a JSON object.
+
+### Entity Specifications
+
+This section details the fields that make up each of the entities referenced
+earlier in this document.
+
+#### `provider` entity
+
+Used in discovery for enumerating the different providers represented in this
+discovery system.
+
+##### `provider` entity attributes
+
+###### provider
+
+- Type: `String`
+- Description: Identifies the provider of the event by nane. The producer string
```

Comment:
I'd avoid the word "Identifies" since that might imply it needs to be unique in some way, which I don't think is your intent. Perhaps something like: A human readable name of the Provider. This attribute does not need to be unique.

Also, you listed it under `provider` but shouldn't be called `name` ?

I wonder if we need to mention which fields can change over time and which MUST remain static. For the purposes of doing updates of a downstream catalog (or just for the consumer to know what's new) we'll need to identify which property is the unique one that MUST NOT change and can be used to uniquely identify it.

Reply:
interesting about things remaining static. If a provider name changes - what happens to the associated subscriptions?